nvidia_triton/datadog_checks/nvidia_triton/check.py

```diff
@@ -0,0 +1,111 @@
+# (C) Datadog, Inc. 2023-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+from urllib.parse import urljoin  # noqa: F401
+
+from six.moves.urllib.parse import urlparse
+
+from datadog_checks.base import AgentCheck, OpenMetricsBaseCheckV2
+from datadog_checks.base.errors import ConfigurationError
+
+from .metrics import METRICS_MAP
+
+DEFAULT_METADATA_ENDPOINT = '/v2'
+DEFAULT_HEALTH_ENDPOINT = DEFAULT_METADATA_ENDPOINT + '/health/ready'
+
+
+class NvidiaTritonCheck(OpenMetricsBaseCheckV2):
+
+    # This will be the prefix of every metric and service check the integration sends
+    DEFAULT_METRIC_LIMIT = 0
+    __NAMESPACE__ = 'nvidia_triton'
+
+    def __init__(self, name, init_config, instances=None):
+        super(NvidiaTritonCheck, self).__init__(name, init_config, instances)
+        if 'openmetrics_endpoint' not in self.instance:
+            raise ConfigurationError(
+                "The 'openmetrics_endpoint' option is missing in the Nvidia Triton integration configuration."
+            )
+        self.openmetrics_endpoint = self.instance["openmetrics_endpoint"]
+        self.tags = self.instance.get('tags', [])
+
+        # Get the API server port if specified, otherwise use the default 8000.
+        self.server_port = str(self.instance.get('server_port', 8000))
+
+        self.collect_server_info = self.instance.get('collect_server_info', True)
+
+        # Get the base url from the openmetrics endpoint and construct the server info API endpoint.
+        if self.collect_server_info is True:
+            self.base_url = None
+            try:
+                parts = urlparse(self.openmetrics_endpoint)
+                # Delete the /metrics from the url
+                self.base_url = parts._replace(path="")
+                # Replace the openmetrics port by the server port
+                self.server_info_api = self.base_url._replace(netloc=parts.hostname + ':' + self.server_port).geturl()
+
+            except Exception as e:
```

Comment:
I'm not sure we need this exception clause. It replaces a nice detailed traceback with a generic message followed only by the message of the exception. We even lose the type of the error, which is useful to plug into google.

Either catch and handle some "expected" exceptions, or let the program crash with maximum information!